lib/sinatra/base.rb

```diff
@@ -1005,7 +1005,8 @@ def process_route(pattern, keys, conditions, block = nil, values = [])
       values += match.captures.map! { |v| force_encoding URI_INSTANCE.unescape(v) if v }
 
       if values.any?
-        original, @params = params, params.merge('splat' => [], 'captures' => values)
+        original = @params
+        @params = @params.merge('splat' => [], 'captures' => values)  # makes a copy of @params
```

Comment:
Yes the merge returns a copy of `@params`.  As opposed to 
<pre><code>@original = @parms.dup
@parms.merge!('splat' => [], 'captures' => values) 
</code></pre>
Which would give the same effect.